DQMOffline/Trigger/python/SiPixel_OfflineMonitoring_TrackCluster_cff.py

```diff
@@ -116,24 +116,160 @@
+  name = "sizex_on_track_inner",
+  title = "Cluster Size along radial vs. prediction (OnTrack) inner ladders",
+)
+
+hltSiPixelPhase1TrackClustersOnTrackSizeXF = hltSiPixelPhase1TrackClustersOnTrackSizeYF.clone(
+  name = "sizex_on_track_forward",
+  title = "Cluster Size radial vs. prediction (OnTrack) forward",
+)
+
+
+
+hltSiPixelPhase1TrackClustersOnTrackSizeXYOuter = hltSiPixelPhase1TrackClustersOnTrackSizeYOuter.clone(
+  name = "sizexy_on_track_outer",
+  title = "Cluster Size x vs y (OnTrack) outer ladders",
+  xlabel = "y size",
+  ylabel = "x size",
+  range_min = 0, range_max  = 20, range_nbins   = 20,
+  range_y_min = 0, range_y_max = 10, range_y_nbins = 10 
+)
+
+hltSiPixelPhase1TrackClustersOnTrackSizeXYInner = hltSiPixelPhase1TrackClustersOnTrackSizeXYOuter.clone(
+ name = "sizexy_on_track_inner",
+ title = "Cluster Size x vs y (OnTrack) inner ladders"
+)
+
+hltSiPixelPhase1TrackClustersOnTrackSizeXYF = hltSiPixelPhase1TrackClustersOnTrackSizeYF.clone(
+  name = "sizexy_on_track_forward",
+  title = "Cluster Size x vs y (OnTrack) forward",
+  xlabel = "y size",
+  ylabel = "x size",
+  range_min = 0, range_max  = 10, range_nbins   = 10,
+  range_y_min = 0, range_y_max = 10, range_y_nbins = 10
+
+)
+
+hltSiPixelPhase1TrackClustersOnTrackChargeOuter = hltDefaultHistoTrack.clone(
+  name = "chargeOuter",
+  title = "Corrected Cluster Charge (OnTrack) outer ladders",
+  range_min = 0, range_max = 150e3, range_nbins = 150,
+  xlabel = "Charge (electrons)",
+
+  specs = VPSet(
+    Specification().groupBy("PXBarrel/PXLayer").save()
+  )
+)
+  
+hltSiPixelPhase1TrackClustersOnTrackChargeInner = hltSiPixelPhase1TrackClustersOnTrackChargeOuter.clone(
+  name = "chargeInner",
+  title = "Corrected Cluster Charge (OnTrack) inner ladders"
+)  
+
+hltSiPixelPhase1TrackClustersOnTrackShapeOuter = hltDefaultHistoTrack.clone(
+  enabled = False,
+  name = "shapeFilterOuter",
+  title = "Shape filter (OnTrack) Outer Ladders",
+  range_min = 0, range_max = 2, range_nbins = 2,
+  xlabel = "shapeFilter",
+  specs = VPSet(
+    Specification().groupBy("PXBarrel/PXLayer").save()
+  )
+)
 
+hltSiPixelPhase1TrackClustersOnTrackShapeInner = hltSiPixelPhase1TrackClustersOnTrackShapeOuter.clone(
+  enabled = False,
+  name = "shapeFilterInner",
+  title = "Shape filter (OnTrack) Inner Ladders",
 )
 
 hltSiPixelPhase1TrackClustersConf = cms.VPSet(
-  hltSiPixelPhase1TrackClustersOnTrackCharge,
-  hltSiPixelPhase1TrackClustersOnTrackSize,
-  hltSiPixelPhase1TrackClustersOnTrackNClusters,
-  hltSiPixelPhase1TrackClustersOnTrackPositionB,
-  hltSiPixelPhase1TrackClustersOnTrackPositionF,
-
-  hltSiPixelPhase1TrackClustersOffTrackCharge,
-  hltSiPixelPhase1TrackClustersOffTrackSize,
-  hltSiPixelPhase1TrackClustersOffTrackNClusters,
-  hltSiPixelPhase1TrackClustersOffTrackPositionB,
-  hltSiPixelPhase1TrackClustersOffTrackPositionF,
-
-  hltSiPixelPhase1TrackClustersNTracks,
-  hltSiPixelPhase1TrackClustersNTracksInVolume,
+### IT HAS TO BE IN SYNCH W/
```

Comment:
hi @mtosi all - is there some technical reason this magic list can not be moved to python, which can be both shared across python modules and communicated to c++ (the reverse, as done now, sharing information in c++ to python, being hard)

Reply:
This is how Marcel design it. the whole pixelPhase1 DQM seems to work in that way.
I do not think it is optimal. I suspect we need to wait for Marcel to be back in CMS and re-design it...

Reply:
sigh- ok, hopefully the fragileness is now documented so that things stay in sync..